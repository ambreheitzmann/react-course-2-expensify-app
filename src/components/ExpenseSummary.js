import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral'; 
import selectTotalExpensesAmount from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';



export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const totalExpensesFormated = numeral(expensesTotal / 100).format('$0,0.00');
    return(
    <div>
            Viewing {expenseCount} expense{expenseCount> 1 && `s`} 
            totalling {totalExpensesFormated}
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibileExpense = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibileExpense.length,
        expensesTotal: selectTotalExpensesAmount(visibileExpense)
    };
  };
  
  export default connect(mapStateToProps)(ExpenseSummary);
