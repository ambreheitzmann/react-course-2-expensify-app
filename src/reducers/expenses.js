// Expenses Reducer

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  console.log(action.type);
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [
        ...state,
        action.expense
      ];
    case 'REMOVE_EXPENSE':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        };
      });
    case 'SET_EXPENSES':
      return action.expenses;
    case 'START_SET_EXPENSES':
      console.log(action.type);
    default:
      return state;
  }
};
