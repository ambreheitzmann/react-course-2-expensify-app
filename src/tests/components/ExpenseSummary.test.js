import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render total 0', () => {
    const wrapper =  shallow(<ExpenseSummary  expenseCount={0} expensesTotal={0}  />);
    expect(wrapper).toMatchSnapshot();
});


test('should render 10 expenses total $100.00', () => {
    const wrapper =  shallow(<ExpenseSummary  expenseCount={10} expensesTotal={10000}  />);
    expect(wrapper).toMatchSnapshot();
});