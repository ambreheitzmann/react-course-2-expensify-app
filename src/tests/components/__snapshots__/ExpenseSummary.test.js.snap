// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 10 expenses total $100.00 1`] = `
<div>
  Viewing 
  10
   expense
  s
  totalling 
  $100.00
</div>
`;

exports[`should render total 0 1`] = `
<div>
  Viewing 
  0
   expense
  totalling 
  $0.00
</div>
`;
